examples/kubernetes/csi-nodeplugin-glusterfsplugin.yaml

```diff
@@ -0,0 +1,83 @@
+# Copyright 2018 The Gluster CSI Authors.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+#You may obtain a copy of the License at
+
+#    http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# This YAML file contains driver-registrar & csi driver nodeplugin API objects
+# that are necessary to run CSI nodeplugin for glusterfs
+kind: DaemonSet
+apiVersion: apps/v1beta2
+metadata:
+  name: csi-nodeplugin-glusterfsplugin
+spec:
+  selector:
+    matchLabels:
+      app: csi-nodeplugin-glusterfsplugin
+  template:
+    metadata:
+      labels:
+        app: csi-nodeplugin-glusterfsplugin
+    spec:
+      serviceAccount: csi-nodeplugin
+      hostNetwork: true
+      containers:
+        - name: driver-registrar
+          image: quay.io/k8scsi/driver-registrar:v0.3.0
+          args:
+            - "--v=5"
+            - "--csi-address=$(ADDRESS)"
+          env:
+            - name: ADDRESS
+              value: /plugin/csi.sock
+            - name: KUBE_NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+          volumeMounts:
+            - name: plugin-dir
+              mountPath: /plugin
+        - name: glusterfs
+          securityContext:
+            privileged: true
+            capabilities:
+              add: ["SYS_ADMIN"]
+            allowPrivilegeEscalation: true
+          image: docker.io/gluster/glusterfs-csi-driver
+          args :
+            - "--nodeid=$(NODE_ID)"
+            - "--endpoint=$(CSI_ENDPOINT)"
+            - "--resturl=http://192.168.121.182:24007"
+            - "--restsecret=b03045b7988258557ecd3e136cd37ba3f928ea0831b3b1b7ed8ae238d36d9071"
+          env:
+            - name: NODE_ID
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+            - name: CSI_ENDPOINT
+              value: unix://plugin/csi.sock
```

Comment:
@Madhu-1 we need to move resturl and restsecret to configmap or secret, we can also put them in env.

Reply:
The secret or configmap enhancement can be in another PR, lets atleast have them in ENV for now.

Reply:
I suggest to leave a comment about the ConfigMap/Secret approach here. The example should be as simple as possible IMHO. Just warn "don't do it like this in production".